server/src/main/java/io/crate/protocols/postgres/PostgresWireProtocol.java

```diff
@@ -867,6 +912,25 @@ private ByteBuf nullOrBuffer(ByteBuf buffer, State nextState) {
             state = nextState;
             return buffer.readBytes(msgLength);
         }
+
+        private State nextState(int requestCode) {
+            // more information on postgres requestCode:
+            // https://www.postgresql.org/docs/current/protocol-message-formats.html
+            switch (requestCode) {
+                case 80877102:
+                    return State.CANCEL_BODY;
+                case 196608:
+                    return State.STARTUP_BODY;
+                default:
+                    throw new IllegalStateException("The given request code is invalid or not yet supported: " + requestCode);
+            }
+        }
     }
 
+    record KeyData(int pid, int secretKey) {
+        static KeyData generate() {
+            long l = ThreadLocalRandom.current().nextLong();
+            return new KeyData((int) (l >> 32), (int) l);
```

Comment:
I still would go with a incrementing `pid` instead of deriving it from the random value. Could be part of `PgSessions` via a `AtomicInteger` .

`dirtyUUID` uses 128 bit, here we have 64 bit. I think that makes quite a difference in terms of potential collisions.

Reply:
I looked at https://www.postgresql.org/docs/current/monitoring-stats.html#MONITORING-PG-STAT-ACTIVITY-VIEW. Shouldn't `pid` be unique per `PostgresWireProtocol` or `Session`? If `pid` is unique per node, isn't it similar to node id?